core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/PhysicalOp.scala

```diff
@@ -345,150 +321,130 @@ case class PhysicalOp(
     this.copy(schemaInfo = Some(schemaInfo))
 
   /**
-    * creates a copy with an additional input operator specified on an input port
+    * creates a copy with the blocking input port indices
     */
-  def addInput(fromOpId: PhysicalOpIdentity, fromPort: Int, toPort: Int): PhysicalOp = {
-    val link = PhysicalLink(fromOpId, fromPort, this.id, toPort)
-    addInput(link)
+  def withBlockingInputs(blockingInputs: List[PortIdentity]): PhysicalOp = {
+    this.copy(blockingInputs = blockingInputs)
   }
 
   /**
     * creates a copy with an additional input operator specified on an input port
     */
   def addInput(link: PhysicalLink): PhysicalOp = {
```

Comment:
rename `addInput` and `addOutput` to `addInputLink` and `addOutputLink`?

Reply:
renamed